### PR TITLE
add json output for odo env view

### PR DESCRIPTION
**Pull request by girishramnani**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind bug

**What does does this PR do / why we need it**:
see $SUBJECT

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/4243

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
```
{
        "kind": "EnvInfo",
        "apiVersion": "odo.dev/v1alpha1",
        "metadata": {
                "creationTimestamp": null
        },
        "spec": {
                "name": "nodejs",
                "project": "aaaa",
                "url": [
                        {
                                "name": "nodejs-3000",
                                "kind": "route"
                        }
                ],
                "appName": "new-app",
                "runMode": "run"
        }
}
```

### CODE REVIEW

**Comment by girishramnani** [comment]
Please dont ask me to write test for this - we wouldn't be testing anything using that test other then json marshaling and flag parsing

File: pkg/envinfo/envinfo.go [hunk 1]
```diff
@@ -60,26 +67,19 @@ const (
...
+	TLSSecret string `yaml:"TLSSecret,omitempty" json:"tlsSecret,omitempty"`
 	// Exposed port number for docker container, required for local scenarios
-	ExposedPort int `yaml:"ExposedPort,omitempty"`
+	ExposedPort int `yaml:"ExposedPort,omitempty" json:"exposedPort,omitempty"`
 	// Kind is the kind of the URL
-	Kind URLKind `yaml:"Kind,omitempty"`
-}
-
-// EnvInfoPushCommand holds the devfile push commands for the component
-type EnvInfoPushCommand struct {
```
Comment (girishramnani) [inline pkg/envinfo/envinfo.go:79]: this struct wasn't used anywhere so removed it

**Review by dharmit — APPROVED** [review]
/approve
/lgtm

Since we're moving towards making just one approval enough for PRs to be merged.

**Pull request merged**